libtest/mock-transport.c

```diff
@@ -188,26 +188,35 @@ log_transport_mock_free_method(LogTransport *s)
   log_transport_free_method(s);
 }
 
+static void
+inject_error(LogTransportMock *self, const gchar *buffer, gssize length)
+{
+  data_t data = {.type = DATA_ERROR, .error_code = GPOINTER_TO_UINT(buffer)};
+  g_array_append_val(self->value, data);
+}
+
+static void
+inject_chunk(LogTransportMock *self, const gchar *buffer, gssize length)
+{
+  if (length == -1)
+    length = strlen(buffer);
+
+  data_t data = {.type = DATA_STRING, .iov = (struct iovec_const)
+  {
+    .iov_base = buffer, .iov_len = length
+  }
+                };
+  g_array_append_val(self->value, data);
+
+}
+
 void
 log_transport_mock_inject_data(LogTransportMock *self, const gchar *buffer, gssize length)
 {
   if (length == LTM_INJECT_ERROR_LENGTH)
```

Comment:
The unfortunate thing is we cannot get rid of LTM_INJECT_ERROR_LENGTH cleanly because the log_transport_mock constructors support injecting data inside the constructor parameters.

```
LogTransport *
log_transport_mock_records_new(const gchar *read_buffer1, gssize read_buffer_length1, ...)
```

In the constructors, we still need to allow injecting error.

Not that this feature is used: otherwise some unit tests should have failed on your earlier comment when I compared -2 to an unsigned variable. So this is probably a dead code now. The reason might be is that LogTransportMock already contains a very lightweight error simulation: each even (2k+1) reads returns an EAGAIN, each odd (2k) reads return a chunk.

Reply:
Thanks for the explanation.

We could get rid of the -2 values here as well if record objects would be used (for example, MockRecord as an interface with 2 implementations: DataMockRecord and ErrorMockRecord), but this is too much work for this PR. Let's just leave it as is. :+1: